### ISSUE TITLE
8$: warn on ASI between string literals

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1020]
```javascript
    let source =
      "ASM_X86_64(\n" //
      "  // xor %eax, %eax\n" + //
      ")\n";
    let expected =
      "ASM_X86_64(\n" //
      "  // xor %eax, %eax\n" + //
      "  0x31, 0xc0,\n" + //
      ")\n";
    assert.strictEqual(await updateTestASMAsync(source), expected);
```
This test passes if `updateTestASMAsync` is the identity function. This is because a semicolon is inserted after each `"ASM_X86_64(\n" //` line, thus `source` and `expected` refer to the same string instead of the intended strings.

**Comment by strager** [comment]
Note: The warning should not trigger for code like this:
```javascript
let s = "hello"
"goodbyte".match(re)
```
